dropwizard-jdbi3-transactions-core/src/main/kotlin/com/campspot/jdbi3/DAOManager.kt

```diff
@@ -3,25 +3,29 @@ package com.campspot.jdbi3
 import org.jdbi.v3.core.Handle
 import kotlin.reflect.KClass
 
-open class DAOManager(private vararg val daos: KClass<out DAO>) {
+open class DAOManager {
   private val transaction = ThreadLocal<Handle>()
   private val daoInstances = ThreadLocal<HashMap<String, DAO>>()
 
   fun setupWithTransaction(transaction: Handle): Handle {
     this.transaction.set(transaction)
-    val daoMap = HashMap<String, DAO>()
-    daos.forEach { dao ->
-      val createdDao = transaction.attach(dao.java)
-      daoMap[dao.qualifiedName ?: "none"] = createdDao
-    }
+    this.daoInstances.set(HashMap())
 
-    daoInstances.set(daoMap)
     return transaction
   }
 
   open operator fun <T: DAO> get(dao: KClass<T>): T {
+    val daoMap = daoInstances.get()
+    var daoInstance = daoMap[dao.qualifiedName]
+    val transactionInstance = transaction.get()
+
+    if (daoInstance == null) {
```

Comment:
If you're feeling frisky, I think you can use the `computeIfAbsent` method for this whole thing?